### DISCUSSION TITLE
Extend header with button(s)

### DISCUSSION THREAD

**Discussion by KoalaBear84** [fiduswriter/Simple-DataTables#358] [Q&A] [answered]
I would like to 'integrate' two simple buttons inside the table header so they don't take extra vertical space in the page.

In a way related to #321

Example:
![image](https://github.com/fiduswriter/simple-datatables/assets/5886900/0a6f7ecc-53f2-4dfa-801b-3a81399be7fd)

Any official options?

My unofficial one (do **NOT** use):
```javascript
document.querySelector('.datatable-search').insertAdjacentHTML('afterend',
	`<div class="float-end">
		<div class="m-1 me-2">
			<button type="button" class="btn btn-sm btn-outline-primary">Button 1</button>
			<button type="button" class="btn btn-sm btn-outline-primary">Button 2</button>
		</div>
	</div>`
);
```

**Comment by johanneswilm** [comment, marked as answer]
Hey, yes this should be easily achievable by specifying a [custom template](https://fiduswriter.github.io/simple-datatables/documentation/template). 

Please don't do what you did "unofficially" there as it is likely to break sdt.

**Reply by KoalaBear84** [reply]
Thanks a lot!

Absolutely, that's why I was looking for an official way. My search for things related to "header" was too specific :)

And was a little confused at first about the template part because of the confusing formatting 😅 But it all made sense after. And found out Visual Studio Code does somewhat the same.

![image](https://github.com/fiduswriter/simple-datatables/assets/5886900/c0a46895-adc4-4201-ae01-777a83fb6f59)
